### ISSUE TITLE
Join endpoint should be versioned and be made more RESTful

### ISSUE THREAD

**Issue by mjvmroz** [floor-thirteen/ddong-server#3]
At present, `GET /join` creates an account. This should probably be more along the lines of `POST /api/1/account`.

**Comment by tristanHessell** [comment]
Agreed. GET was used for testing (and mistakenly not changed back to POST).
